### PR TITLE
Set of improvements/fixes in Gridap.Adaptivity module

### PR DESCRIPTION
**Pull request by JordiManyer**
Things part of this PR:  

- [x] New: Implement `get_face_subface_ldof_to_cell_ldof`, a glue between coarse and fine lagrangian dofs for `RedRefinementRules`. 
- [x] Bugfix: Correct labeling of refined models. 
- [x] Bring fixes/changes from GridapP4est.jl#p4est-migration-exploring-non-conformity
- [x] Implement generic version of `get_d_to_face_to_parent_face`
- [x] Implement generic version of `get_d_to_face_to_child_faces`
- [x] Bugfix: Allow domain limits to be floats in [cartesian refinement routines](https://github.com/gridap/Gridap.jl/blob/6ce09af425a15b5a38e6712fec86805ecd6fd4a6/src/Adaptivity/AdaptedDiscreteModels.jl#L123).

### CODE REVIEW

**Comment by JordiManyer** [comment]
@amartinhuertas Have a look at the concept version, experiment and check if it's indeed what you need. If so, I'll implement a generic version for any degree pair `(p,2*p)`. 
I leave a sketch of the toy model I'm using for the concept implementation: 
[DOF_glue.pdf](https://github.com/gridap/Gridap.jl/files/10985345/DOF_glue.pdf)

**Comment by JordiManyer** [comment]
Note that I can also produce very easily the inverse map, i.e for each coarse edge a vector containing the properly ordered dof ids of the fine edges below it.

**Comment by JordiManyer** [comment]
The CI errors are weird.... I have not touched that part of the code, and all tests pass locally.... Could this be a mistake in the master branch?

File: src/Adaptivity/RefinementRules.jl [hunk 1]
```diff
@@ -104,6 +104,92 @@ function bundle_points_by_subcell(rr::RefinementRule,x::AbstractArray{<:Point})
...
+      return [orders[mask]...]
+    end
+  else
+    @notimplemented
+  end
+
+  return face_orders
+end
+
+function coarse_nodes_above_fine_nodes(rr::RefinementRule{ExtrusionPolytope{Dc}},
```
Comment (amartinhuertas) [inline src/Adaptivity/RefinementRules.jl:149]: wouldnt be a name such as:

`get_face_subface_ldof_to_cell_ldof` 

more appropriate? just a suggestion ...

Comment (JordiManyer) [reply src/Adaptivity/RefinementRules.jl:149]: Sure! I'll get that changed.

Comment (JordiManyer) [reply src/Adaptivity/RefinementRules.jl]: Resolved by [8bfc6f3](https://github.com/gridap/Gridap.jl/pull/886/commits/8bfc6f3c7c7f4948843794f0d97542b4a9e28d98)

**Comment by JordiManyer** [comment]
For future reference, here are the refinement rules for HEX: 

image

image

image

image

File: src/Adaptivity/OldToNewFields.jl [hunk 2]
```diff
@@ -0,0 +1,61 @@
+"""
```
Comment (amartinhuertas) [inline src/Adaptivity/OldToNewFields.jl:1]: Update comment ...

**Pull request merged**